File: MorySnip/Forms/Form_Download.vb
Public Class Form_Download
    Public Property ShareId() As String
    Public Property ItemsCount() As Integer

    Public Sub BeginDownload()
        Me.ProgressBar_Files.Maximum = Me.ItemsCount

        For i As Integer = 0 To Me.ItemsCount - 1
            Do
                Try
                    Me.Download(Me.ShareId & i.ToString("000"))
                    Exit Do
                Catch ex As Exception
                    Select Case MsgBox(ex.Message, MsgBoxStyle.Critical Or MsgBoxStyle.AbortRetryIgnore)
                        Case MsgBoxResult.Retry
                        Case MsgBoxResult.Abort
                            Exit For
                        Case MsgBoxResult.Ignore
                            Exit Do
                    End Select
                End Try
            Loop
            Me.ProgressBar_Files.Value = i
        Next

        Me.Close()
    End Sub

    Private Sub Download(Id As String)
        My.Computer.Network.DownloadFile(String.Format("http://share.cucak.am/snapshots/{0}.png", Id), String.Format("{1}\tmp\{0}.png", Id, My.Computer.FileSystem.SpecialDirectories.Desktop), "", "", True, 3000, True, FileIO.UICancelOption.ThrowException)
    End Sub

    Private Sub Form_Download_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub

    Private Sub Timer_Begin_Tick(sender As Object, e As EventArgs) Handles Timer_Begin.Tick
        Me.Timer_Begin.Stop()
        Me.BeginDownload()
    End Sub
End Class

File: MorySnip/Forms/Form_Download.cs
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.Devices;
using System;

namespace Morysoft.MorySnip
{
    public partial class Form_Download
    {
        public string ShareId { get; set; }
        public int ItemsCount { get; set; }

        public void BeginDownload()
        {
            this.ProgressBar_Files.Maximum = this.ItemsCount;

            for (int i = 0, loopTo = this.ItemsCount - 1; i <= loopTo; i++)
            {
                do
                {
                    try
                    {
                        this.Download(this.ShareId + i.ToString("000"));
                        break;
                    }
                    catch (Exception ex)
                    {
                        switch (Interaction.MsgBox(ex.Message, MsgBoxStyle.Critical | MsgBoxStyle.AbortRetryIgnore))
                        {
                            case MsgBoxResult.Retry:
                                {
                                    break;
                                }

                            case MsgBoxResult.Abort:
                                {
                                    break;
                                    break;
                                }

                            case MsgBoxResult.Ignore:
                                {
                                    break;
                                    break;
                                }
                        }
                    }
                }
                while (true);
                this.ProgressBar_Files.Value = i;
            }

            this.Close();
        }

        private void Download(string Id)
        {
            Network.DownloadFile(string.Format("http://share.cucak.am/snapshots/{0}.png", Id), string.Format(@"{1}\tmp\{0}.png", Id, System.IO.Path.des.SpecialDirectories.Desktop), "", "", true, 3000, true, Microsoft.VisualBasic.FileIO.UICancelOption.ThrowException);
        }

        private void Form_Download_Load(object sender, EventArgs e)
        {
        }

        private void Timer_Begin_Tick(object sender, EventArgs e)
        {
            this.Timer_Begin.Stop();
            this.BeginDownload();
        }
    }
}
